### PR TITLE
Initialize SQlite database for realtime collab

### PR DESCRIPTION
**Pull request by navilperez**

### CODE REVIEW

File: blockly-rtc/src/db.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview The SQlite database object.
+ * @author navil@google.com (Navil Perez)
+ */
+const sqlite3 = require('sqlite3');
+
+let db = new sqlite3.Database("./events.sqlite", function(error) {
```
Comment (samelhusseini) [inline blockly-rtc/src/db.js]: const

File: blockly-rtc/src/db_utils.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Util scripts for interacting with the database.
```
Comment (samelhusseini) [inline blockly-rtc/src/db_utils.js:19]: Add @author

File: blockly-rtc/src/db_utils.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ */
+
+var db = require('./db');
```
Comment (samelhusseini) [inline blockly-rtc/src/db_utils.js]: const

File: blockly-rtc/src/db_utils.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+/**
+ * Query the database for rows since the given server id.
+ * @param {number} serverId serverId for the lower bound of the query.
+ * @return {!Array.<!Object>} The rows since the last given server id.
+ * @public
+ */
+function queryDatabase(serverId) {
+  var sql = `SELECT * from events WHERE serverId > ${serverId};`;
+  db.all(sql, [], (err, rows) => {
```
Comment (samelhusseini) [inline blockly-rtc/src/db_utils.js]: You can probably lose the ``sql`` variable here.

File: blockly-rtc/src/db_utils.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+/**
+ * Add rows to the database.
+ * Rows are added transactionally.
+ * @param {!Array.<!Object>} rows The rows to be added to the database.
+ * @public
+ */
+function addToServer(rows) {
+  var insertQueries = [];
```
Comment (samelhusseini) [inline blockly-rtc/src/db_utils.js]: const

File: blockly-rtc/src/db_utils.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+    db.run('COMMIT;');
+  });   
+
+};
+
+/**
+ * Create a SQlite query script to insert a new row.
+ * @param {!Object} row serverId for the lower bound of the query.
+ * @return {!String} The SQlite query.
```
Comment (samelhusseini) [inline blockly-rtc/src/db_utils.js]: lowercase string and strings are non-nullable by default, so ``{string}``

**Review by samelhusseini — COMMENTED** [review]
What's in the binary? Does it need to be checked in or can it be generated by the service when it starts?

**Comment by navilperez** [comment]
> What's in the binary? Does it need to be checked in or can it be generated by the service when it starts?

It is the binary for the database since I am using an in-disk database. I just ran the node server without it and it seems as if the service can generate it, so it does not need to be checked in. I will go ahead and remove it. Thank you for drawing my attention to that.

File: blockly-rtc/src/db.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+/**
+ * @fileoverview The SQlite database object.
+ * @author navil@google.com (Navil Perez)
+ */
+
+const sqlite3 = require('sqlite3');
+
+const db = new sqlite3.Database("./events.sqlite", (err) => {
```
Comment (samelhusseini) [inline blockly-rtc/src/db.js]: Following the style guide, use single quotes here.

**Review by samelhusseini — COMMENTED** [review]
One last nit but otherwise lgtm. My preference would be for this to go into a branch, until the demo the ready for release. Placing these files there on there own might confuse people using this samples repo, especially without a README.

**Review by samelhusseini — APPROVED** [review]

**Comment by samelhusseini** [comment]
LGTM.

**Pull request merged**